src/imager.cpp

```diff
@@ -120,12 +128,27 @@ auto Imager<T>::collect(T wl, const std::function<void(std::size_t, std::size_t,
                         ConstView<T, 2> xyz, ConstView<T, 2> uvw) -> void {
   if (!synthesis_) throw InternalError();
 
+  auto& ctx = *synthesis_->context();
+
+  // Count the number of non-zero visibilities
+  assert(s.shape(0) == s.shape(1));
+  std::size_t nVis = {0};
+  const std::complex<T> c0 = 0.0;
+  nVis = s.shape(0) * s.shape(1);
+  const auto S = s.data();
```

Comment:
Anything that's just a `View` and not a `HostView` can be located on either device or host memory. You'd want to use a `DeviceAccessor` to generate a `HostView` as needed. But as commented before, I'd suggest moving this to the eigensolver, where we already iterate through this matrix in host memory.

Reply:
Moved to the eigensolver.